### PR TITLE
feat: get blocks from runme binary

### PR DESCRIPTION
**Pull request by duvanmonsa**
Adding changes to the launcher to use runner

### CODE REVIEW

File: src/extension/runner.ts [hunk 1]
```diff
@@ -244,6 +248,34 @@ export class GrpcRunner implements IRunner {
...
+          return blocks
+        })
+        .catch((e) => {
+          throw e
+        })
+    } catch (err: any) {
+      console.error(err)
+      this.close()
+      throw err
+    }
```
Comment (christian-bromann) [inline src/extension/runner.ts:276]:
```suggestion
      const { response } = await client.getBlocks(request)
      return response.blocks || []
    } catch (err: any) {
      console.error(err)
      this.close()
      throw err
    }
```

File: src/extension/provider/launcher.ts [hunk 2]
```diff
@@ -58,10 +59,16 @@ export class RunmeFile extends TreeItem {
 export class RunmeLauncherProvider implements TreeDataProvider<RunmeFile>, Disposable {
   #disposables: Disposable[] = []
 
+  protected runner?: IRunner
+
```
Comment (christian-bromann) [inline src/extension/provider/launcher.ts:63]:
```suggestion
```

File: src/extension/provider/launcher.ts [hunk 3]
```diff
@@ -58,10 +59,16 @@ export class RunmeFile extends TreeItem {
...
   private filesTree: Map<string, TreeFile> = new Map()
   private defaultItemState: TreeItemCollapsibleState = TreeItemCollapsibleState.Collapsed
 
-  constructor(private workspaceRoot?: string | undefined) {
+  constructor(
+    runner: IRunner | undefined,
```
Comment (christian-bromann) [inline src/extension/provider/launcher.ts:68]: I think there is never a case where we don't provide the runner, right?
```suggestion
    protected runner: IRunner,
```

File: src/extension/provider/launcher.ts [hunk 4]
```diff
@@ -156,6 +163,17 @@ export class RunmeLauncherProvider implements TreeDataProvider<RunmeFile>, Dispo
     this.refresh()
   }
 
+  private async loadBlocks() {
+    try {
+      const blocks = await this.runner?.getBlocks()
```
Comment (christian-bromann) [inline src/extension/provider/launcher.ts:168]:
```suggestion
      const blocks = await this.runner.getBlocks()
```

File: src/extension/provider/launcher.ts (hunk 3 above)
Comment (duvanmonsa) [reply src/extension/provider/launcher.ts:68]: I was trying to follow this example here.
https://github.com/stateful/vscode-runme/blob/b242c5e71a39d8cd4063c2f7a20ce2eb9fa9deca/src/extension/extension.ts#L62

**Comment by sourishkrout** [comment]
Closing since we now have https://github.com/stateful/runme/pull/438 and subsequent work on the project service.

**Pull request closed without merging**